src/subsystems/nodejs/builders/granular/tsconfig-to-json.js

```diff
@@ -1,19 +1,21 @@
 try {
-  console.log(require.resolve("typescript"));
-} catch(e) {
-  console.error("typescript is not found");
-  process.exit(e.code);
+  require.resolve("typescript");
+} catch (e) {
+  process.exit(0);
 }
 
-const ts = require("typescript")
-const fs = require('fs')
+const ts = require("typescript");
+const fs = require("fs");
 
 try {
-  const data = fs.readFileSync('tsconfig.json', 'utf8')
+  const data = fs.readFileSync("tsconfig.json", "utf8");
 } catch (err) {
-  console.error(err)
+  console.error(err);
 }
 
-config = ts.parseConfigFileTextToJson(data)
-newdata = JSON.stringify(config)
-fs.writeFileSync('tsconfig.json', newdata);
+config = ts.parseConfigFileTextToJson(data);
+
+// https://www.typescriptlang.org/tsconfig#preserveSymlinks
+config.compilerOptions.preserveSymlinks = true;
```

Comment:
This logic is redundant with https://github.com/nix-community/dream2nix/blob/3223d29754c1ba1223c848e274cd4481c2bcd979/src/subsystems/nodejs/builders/granular/default.nix#L415